src/library/scala/util/Random.scala

```diff
@@ -70,6 +70,16 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextLong(): Long = self.nextLong()
 
+  def nextLong(n: Long): Long = {
+    if (n <= 0) throw new java.lang.IllegalArgumentException("bound must be positive")
+    if (n <= Int.MaxValue) nextInt(n.toInt).toLong
+    else {
+      val RAND_MAX = Long.MaxValue     // 2^63-1
+      val x = nextLong() >>> 1         // [0, 2^63), strip the sign bit
+      if (x < (RAND_MAX - (RAND_MAX % n))) x % n else nextLong(n)
```

Comment:
I like that proposed modification. As to efficiency of div/mod, I'm afraid I don't know a think about scala.js architecture. I wrote this for my general scala use. Since scala.js already has a method for this, would it still be ok to pursue this for inclusion into the scala base itself?